units/host_termination.go

```diff
@@ -202,29 +202,34 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 	} else {
 		// Consider if the host is in between running a single-host task group
 		if j.host.LastGroup != "" {
-			lastTask, err := task.FindOneId(j.host.LastTask)
+			latestTask, err := task.FindOneId(j.host.LastTask)
 			if err != nil {
 				j.AddError(errors.Wrapf(err, "finding last task '%s'", j.host.LastTask))
 			}
 			// Only try to restart the task group if it was successful and should have continued executing.
-			if lastTask != nil && lastTask.IsPartOfSingleHostTaskGroup() && lastTask.Status == evergreen.TaskSucceeded {
-				tasks, err := task.FindTaskGroupFromBuild(lastTask.BuildId, lastTask.TaskGroup)
+			if latestTask != nil && latestTask.IsPartOfSingleHostTaskGroup() && latestTask.Status == evergreen.TaskSucceeded {
+				tasks, err := task.FindTaskGroupFromBuild(latestTask.BuildId, latestTask.TaskGroup)
 				if err != nil {
-					j.AddError(errors.Wrapf(err, "getting task group for task '%s'", lastTask.Id))
+					j.AddError(errors.Wrapf(err, "getting task group for task '%s'", latestTask.Id))
 					return
 				}
 				if len(tasks) == 0 {
-					j.AddError(errors.Errorf("no tasks found in task group for task '%s'", lastTask.Id))
+					j.AddError(errors.Errorf("no tasks found in task group for task '%s'", latestTask.Id))
 					return
 				}
-				if tasks[len(tasks)-1].Id != lastTask.Id {
+				var lastTaskGroupTask task.Task
+				for _, t := range tasks {
+					if t.HasOrWillRun() {
+						lastTaskGroupTask = t
+					}
+				}
```

Comment:
Can you please add a comment to this explaining what it's looking for?

Reply:
Yes!